KTL.js

```diff
@@ -5620,6 +5622,145 @@ function Ktl($, appInfo) {
+        * @param {object} data - records in table*/
+        function stickyTableHeader(viewId, keywords, data) {
+            const kw = '_sth';
+            let numOfRecords = 10;
+            let viewHeight = 800;
+            let isCalledInsideObserver = false;
+
+            if (!keywords[kw]) return;
+
+            if (keywords[kw].length) {
+                numOfRecords = keywords[kw][0].params[0][0] || numOfRecords;
+                viewHeight = keywords[kw][0].params[0][1] || viewHeight;
+            }
+
+            const callStickTableHeaders = () => {
+                if (!isCalledInsideObserver) {
+                    ktl.views.stickTableHeader(viewId, data, numOfRecords, viewHeight);
+                }
+            }
+
+            if (isBulkOpsEnabled(keywords, viewId)) {
+                let isCalledInsideObserver = false;
+                let observerPromise = new Promise((resolve) => {
+                    observeAndExecute(`#bulkOpsControlsDiv-${viewId}`, () => {
+                        ktl.views.stickTableHeader(viewId, data, numOfRecords, viewHeight);
+                        isCalledInsideObserver = true;
+                        resolve();
+                    });
+                });
+
+                const callStickTableHeadersIfNotResolved = () => {
+                    if (!isCalledInsideObserver) {
+                        callStickTableHeaders();
+                    }
+                };
+
+                // If observerPromise does not resolve within 1000ms, call callStickTableHeadersIfNotResolved
+                setTimeout(callStickTableHeadersIfNotResolved, 1000);
+            } else {
+                callStickTableHeaders();
+            }
+        }
+
+        /** _stc = Sticky Table Columns
+        * keyword @param stickyColBkgdColor - The background color of the sticky column
+        * @param {string} viewId - The view.key
+        * @param {object} keywords - The keywords object*/
+        function stickyTableColumns(viewId, keywords) {
+            const kw = '_stc';
+            let stickyColBkgdColor = 'rgb(243 246 249)';
+            let numOfColumns = 1;
+            let isCalledInsideObserver = false;
+
+            if (!keywords[kw]) return;
+
+            if (keywords[kw].length) {
+                numOfColumns = keywords[kw][0].params[0][0] || numOfColumns;
+                stickyColBkgdColor = keywords[kw][0].params[0][1] || stickyColBkgdColor;
+            }
+
+            const callStickTableColumns = () => {
+                if (!isCalledInsideObserver) {
+                    ktl.views.stickTableColumns(viewId, numOfColumns, stickyColBkgdColor);
+                }
+            }
+
+            if (isBulkOpsEnabled(keywords, viewId)) {
+                numOfColumns = Math.max(numOfColumns, 2);
+                let observerPromise = new Promise((resolve) => {
+                    observeAndExecute(`#bulkOpsControlsDiv-${viewId}`, () => {
```

Comment:
After reading `observeAndExecute`,  the 'AndExecute' is misleading. It implies and end to the function call stack. It is more common to name functions like this one `registerObserver` or `registerCallback`. It tells a programmer that the code will be executed more than once and probably forever if there is no 'unregister' .